packages/cosmos-bin/src/cw3.rs

```diff
@@ -0,0 +1,434 @@
+    #[clap(long)]
+    title: String,
+    /// Description, defaults to title
+    #[clap(long)]
+    description: Option<String>,
+    /// Messages, in JSON format
+    #[clap(long)]
+    msg: Vec<String>,
+}
+
+async fn propose(
+    cosmos: Cosmos,
+    ProposeOpt {
+        cw3,
+        tx_opt,
+        title,
+        description,
+        msg,
+    }: ProposeOpt,
+) -> Result<()> {
+    let wallet = tx_opt.get_wallet(cosmos.get_address_type());
+    let cw3 = cosmos.make_contract(cw3);
+    let res = cw3
+        .execute(
+            &wallet,
+            vec![],
+            cw3_flex_multisig::msg::ExecuteMsg::Propose {
+                description: description.unwrap_or_else(|| title.clone()),
+                title,
+                msgs: msg
+                    .into_iter()
+                    .map(|x| {
+                        serde_json::from_str::<CosmosMsg<Empty>>(&x)
+                            .context("Invalid CosmosMsg provided")
+                    })
+                    .collect::<Result<_>>()?,
+                latest: None,
+            },
+        )
+        .await?;
+    log::info!("Added in {}", res.txhash);
+    Ok(())
+}
+
+#[derive(clap::Parser)]
+struct ListOpt {
+    /// CW3 group contract address
+    #[clap(long)]
+    cw3: Address,
+}
+
+async fn list(cosmos: Cosmos, ListOpt { cw3 }: ListOpt) -> Result<()> {
+    let cw3 = cosmos.make_contract(cw3);
+    let mut start_after = None;
+    loop {
+        let ProposalListResponse::<Empty> { proposals } = cw3
+            .query(cw3_flex_multisig::msg::QueryMsg::ListProposals {
+                start_after,
+                limit: None,
+            })
+            .await?;
+        match proposals.last() {
+            None => break Ok(()),
+            Some(proposal) => start_after = Some(proposal.id),
+        }
+        for ProposalResponse {
+            id,
+            title,
+            description: _,
+            msgs: _,
+            status,
+            expires: _,
+            threshold: _,
+            proposer: _,
+            deposit: _,
+        } in proposals
+        {
+            println!("{id}: {title}. {status:?}");
+        }
+    }
+}
+
+#[derive(clap::Parser)]
+struct VoteOpt {
+    #[clap(flatten)]
+    tx_opt: TxOpt,
+    /// CW3 group contract address
+    #[clap(long)]
+    cw3: Address,
+    /// Proposal ID to execute
+    #[clap(long)]
+    proposal: u64,
+    /// How to vote
+    #[clap(long)]
+    vote: String,
+}
+
+async fn vote(
+    cosmos: Cosmos,
+    VoteOpt {
+        tx_opt,
+        cw3,
+        proposal,
+        vote,
+    }: VoteOpt,
+) -> Result<()> {
+    let wallet = tx_opt.get_wallet(cosmos.get_address_type());
+    let cw3 = cosmos.make_contract(cw3);
+    let res = cw3
+        .execute(
+            &wallet,
+            vec![],
+            cw3_flex_multisig::msg::ExecuteMsg::Vote {
+                proposal_id: proposal,
+                vote: serde_json::from_value(serde_json::Value::String(vote))?,
```

Comment:
do we want to disallow abstaining or veto?

Reply:
I don't think so, can you think of a reason to disallow it?